core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java

```diff
@@ -410,6 +422,40 @@ public BlockReader createFileReader(String fileId, long offset, boolean position
         mPageSize);
   }
 
+  @Override
+  public ListenableFuture<List<FileFailure>> load(List<File> files) {
+    List<ListenableFuture<Void>> futures = new ArrayList<>();
+    List<FileFailure> errors = Collections.synchronizedList(new ArrayList<>());
+    for (File file : files) {
+      ListenableFuture<Void> loadFuture = mIOExecutor.submit(() -> {
+        try {
+          load(file.getUfsPath(), file.getMountId(), file.getLength());
+        } catch (Exception e) {
+          AlluxioRuntimeException t = AlluxioRuntimeException.from(e);
+          errors.add(FileFailure.newBuilder().setFile(file).setCode(t.getStatus().getCode().value())
+                                .setMessage(t.getMessage()).build());
+        }
+        return null;
+      });
+      futures.add(loadFuture);
+    }
+    ListenableFuture<List<Void>> future = Futures.successfulAsList(futures);
+    return Futures.transform(future, input -> errors, mIOExecutor);
+  }
+
+  private void load(String ufsPath, long mountId, long length) {
+    Protocol.OpenUfsBlockOptions options =
+        Protocol.OpenUfsBlockOptions.newBuilder().setUfsPath(ufsPath).setMountId(mountId)
+                                    .setNoCache(false).setOffsetInFile(0).setBlockSize(length)
+                                    .build();
+    String fileId = new AlluxioURI(ufsPath).hash();
+    try (BlockReader fileReader = createFileReader(fileId, 0, false, options)) {
+      fileReader.read(0, length);
```

Comment:
the buffer size is equal to the file length. Can we use a smaller buffer and do multiple rounds of read to further reduce the memory pressure?